File: 2015-08-15-Python与Matlab科学计算比较/calcuateBigMatrix.m
close;
clear all;

matrixRow = 1000;
matrixColumn = 2000;

tic

a = 0:1:matrixRow*matrixColumn-1;
b = 0:1:matrixRow*matrixColumn-1;
a = reshape(a, matrixRow, matrixColumn);
b = reshape(b, matrixColumn, matrixRow);
c = a * b;

size(c)

toc

File: 2015-08-15-Python与Matlab科学计算比较/calcuateBigMatrix.py
#coding=utf-8
import numpy as np
import time


start = time.clock()

matrixRow = 1000
matrixColumn = 2000
a = np.arange(matrixRow*matrixColumn).reshape(matrixRow, matrixColumn)
b = np.arange(matrixRow*matrixColumn).reshape(matrixColumn, matrixRow)
c = np.dot(a, b)
print c.shape

end = time.clock()
print "spend time: %f s" % (end - start)